### PR TITLE
Support proxy for Docker Hub

### PR DESCRIPTION
**Pull request by lentzi90**
We already have `CONTAINER_REGISTRY` for setting the "main" registry to use (basically quay.io proxy). With this PR we can also set `DOCKER_HUB_PROXY` to use a different proxy for Docker Hub. The default is `docker.io`, which basically means no change.

For the registry image, we used to just specify `registry`. I changed this to include the Docker Hub proxy and set the path to `library/registry` since this is the "real" image on Docker Hub. It defaults to `library/` if nothing is specified on Docker Hub, but this does not work well with proxies.

### CODE REVIEW

**Review by Rozzii — APPROVED** [review]

File: 03_launch_mgmt_cluster.sh [hunk 1]
```diff
@@ -500,21 +500,15 @@ create_clouds_yaml
...
   patch_clusterctl
   launch_cluster_api_provider_metal3
-fi
 
-if [ "${EPHEMERAL_CLUSTER}" != "tilt" ]; then
     if [ "${CAPM3_VERSION}" != "v1alpha4" ]; then
         launch_baremetal_operator
     fi
     launch_ironic
-fi
```
Comment (fmuyassarov) [inline 03_launch_mgmt_cluster.sh:515]: I think we can improve a bit further 
```shell
if [ "${EPHEMERAL_CLUSTER}" != "tilt" ]; then
  start_management_cluster
  kubectl create namespace metal3

  patch_clusterctl
  launch_cluster_api_provider_metal3

    if [ "${CAPM3_VERSION}" == "v1alpha4" ]; then
        BMO_NAME_PREFIX="${NAMEPREFIX}-baremetal-operator"
    else
        BMO_NAME_PREFIX="${NAMEPREFIX}"
        launch_baremetal_operator
    fi
    launch_ironic

```

**Review by fmuyassarov — APPROVED** [review]

File: 03_launch_mgmt_cluster.sh (hunk 1 above)
Comment (lentzi90) [reply 03_launch_mgmt_cluster.sh:515]: Good point, thanks! Will update :slightly_smiling_face:

**Review by furkatgofurov7 — COMMENTED** [review]
/lgtm

**Pull request merged**